units/provisioning_setup_host.go

```diff
@@ -805,33 +799,25 @@ func mountLinuxVolume(ctx context.Context, env evergreen.Environment, h *host.Ho
 		}))
 	}()
 
-	device, err := getMostRecentlyAddedDevice(ctx, env, h)
+	// wait for the volume to come up on the instance
+	device, err := waitForDevice(ctx, env, h)
 	if err != nil {
-		return errors.Wrap(err, "can't get device info")
+		return errors.Wrap(err, "can't get device")
 	}
 
-	cmd := client.CreateCommand(ctx).Sudo(true)
-	// umount if it's already mounted
+	// we've done this already
 	if device.MountPoint != "" {
-		cmd.Append(fmt.Sprintf("umount -f /dev/%s", device.Name))
+		return nil
 	}
-	// format if the volume doesn't have a filesystem
+
+	// this is a fresh volume
 	if device.FSType == "" {
```

Comment:
Out of curiosity, why does fstype determine the new/existing volume? I don't actually know what fstype implies about the volume.

Reply:
if it's a volume we just got from AWS it will never have had a filesystem created on it. If we or someone else has once created a filesystem I wanted to avoid making changes on their volume.